test/allowlist_for_publicAPI.json

```diff
@@ -449,6 +450,7 @@
     "DebugLevel",
     "Dict",
     "GatherOptions",
+    "namedtuple",
```

Comment:
We're transition c10d to be fully typed with https://github.com/pytorch/pytorch/pull/84119